base-containers/base/inginious_container_api/utils.py

```diff
@@ -27,8 +27,9 @@ def set_limits_user(user):
 
 
 def set_executable(filename):
-    st = os.stat(filename)
-    os.chmod(filename, st.st_mode | stat.S_IEXEC)
+    if "course/common" not in filename:
```

Comment:
At this moment, we should add why in comment (because the folder is mounted readonly in the container). A _startswith_ check with the complete prefix, instead of a _contains_ check, would ensure this is only applied to the concerned files.

I would investigate, in a future patch, about moving the whole ``set_executable`` thing to the agent where the temporary folder is created. (This can also be mentioned in the comment to keep track)